homeassistant/components/zamg/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""Data Update coordinator for ZAMG weather data."""
+from __future__ import annotations
+
+from zamg import ZamgData as ZamgDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_STATION_ID, DOMAIN, LOGGER, MIN_TIME_BETWEEN_UPDATES
+
+
+class ZamgDataUpdateCoordinator(DataUpdateCoordinator[ZamgDevice]):
+    """Class to manage fetching ZAMG weather data."""
+
+    config_entry: ConfigEntry
+    data: dict = {}
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        entry: ConfigEntry | None = None,
+        station_id: str | None = None,
+    ) -> None:
+        """Initialize global ZAMG data updater."""
+        self.zamg = ZamgDevice(session=async_get_clientsession(hass))
+        if entry:
+            self.zamg.set_default_station(entry.data[CONF_STATION_ID])
+        if station_id:
+            self.zamg.set_default_station(station_id)
```

Comment:
```suggestion
    ) -> None:
        """Initialize global ZAMG data updater."""
        self.zamg = ZamgDevice(session=async_get_clientsession(hass))
        self.zamg.set_default_station(entry.data[CONF_STATION_ID])
```
This is nowhere initialized with `station_id` attribute.

Reply:
yes right..